src/Tools/dotnet-openapi/src/Commands/RefreshCommand.cs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Build.Evaluation;
+using Microsoft.Extensions.CommandLineUtils;
+using Microsoft.Extensions.Tools.Internal;
+
+namespace Microsoft.DotNet.OpenApi.Commands
+{
+    internal class RefreshCommand : BaseCommand
+    {
+        private const string CommandName = "refresh";
+
+        public RefreshCommand(Application parent) : base(parent, CommandName)
+        {
+            _sourceFileArg = Argument(SourceProjectArgName, $"The openapi reference to refresh.");
+        }
+
+        internal readonly CommandArgument _sourceFileArg;
+
+        protected override async Task<int> ExecuteCoreAsync()
+        {
+            var projectFile = ResolveProjectFile(ProjectFileOption);
+
+            var sourceFile = Ensure.NotNullOrEmpty(_sourceFileArg.Value, SourceProjectArgName);
+
+            if (IsUrl(sourceFile))
+            {
+                var destination = FindReferenceFromUrl(projectFile, sourceFile);
+
+                await DownloadToFileAsync(sourceFile, destination, overwrite: true);
+            }
+            else
+            {
+                throw new ArgumentException($"'dotnet openapi refresh' must be given a URL");
+            }
+
+            return 0;
+        }
+
+
+        private string FindReferenceFromUrl(FileInfo projectFile, string url)
+        {
+            var project = LoadProject(projectFile);
+            var openApiReferenceItems = project.GetItems(OpenApiReference);
+
+            foreach (ProjectItem item in openApiReferenceItems)
```

Comment:
If you are already using foreach, might as well as go all in on Linq and use .Where()

Reply:
That seems a bit hefty for a Linq query.

Reply:
How is it hefty? optionApiReferenceItems.Single/First(i => string.Equals(i.GetMetadataValue(SourceUrlAttrName), url, StringComparison.Ordinal)) is a lot concise than the loop.

Reply:
That's not the complete logic it would be replacing though. To do it your way it would have to be:
```c#
var ref = openApiReferenceItems.FirstOrDefault(i => string.Equals(i.GetMetadataValue(SourceUrlAttrName), url, StringComparison.Ordinal))?.EvaluatedInclude;

if(ref == null)
{
    throw new ...
}
else{
   return ref;
}
```

Your way is certainly shorter in terms of LOC, but I think mine is easier to follow.

Reply:
Hehe, null coalescing operators 
```
optionApiReferenceItems.FirstOrDefault(i => string.Equals(i.GetMetadataValue(SourceUrlAttrName), url, StringComparison.Ordinal)) ?? throw new ArgumentException(...);
```

Just playing code golf at this point. Not gonna hold up this PR for it :)